.github/workflows/ci.yml

```diff
@@ -0,0 +1,119 @@
+name: CI
+
+on:
+  pull_request:
+    branches: [ independent-releases ]
+  push:
+    branches: [ independent-releases ]
+  workflow_dispatch:
+
+jobs:
+  changes:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: read
+    outputs:
+      packages: ${{ steps.filter.outputs.changes }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dorny/paths-filter@v3
+        id: filter
+        with:
+          filters: |
+            scan-docker-image: 'security-actions/scan-docker-image/**'
+            sca: 'security-actions/sca/**'
+            scan-rust: 'security-actions/scan-rust/**'
+            semgrep: 'security-actions/semgrep/**'
+            sign-docker-image: 'security-actions/sign-docker-image/**'
+            code-build-actions: 'code-build-actions/build-js-sdk/**'
+            lua-lint: 'code-check-actions/lua-lint/**'
+            rust-lint: 'code-check-actions/rust-lint/**'
+            slack-actions: 'slack-actions/workflow-notification/**'
+            cleanup: 'pr-previews/cleanup/**'
+            up-to-date: 'pr-previews/up-to-date/**'
+            validate: 'pr-previews/validate/**'
+
+  setup-dependencies:
+    needs: changes
+    runs-on: ubuntu-latest
+    if: ${{ needs.changes.outputs.packages != '[]' }}
+    outputs:
+      nodeVersion: ${{ steps.node-version.outputs.nodeVersion }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Setup Node
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+
+      - name: Node Version
+        id: node-version
+        run: |
+          nodeVersion="$(node --version | grep -oE '[0-9]+' | head -n1)"
+          echo "nodeVersion=$nodeVersion" >> $GITHUB_OUTPUT
+
+      - name: Install PNPM
+        run: |
+          npm install -g pnpm
+          pnpm --version
+          if [ -n "${{ github.head_ref }}" ]; then
+            git checkout ${{ github.head_ref }}
+          fi
+
+      - name: Dependency Cache
```

Comment:
Caching pnpm `node_modules (optimizes build speed but might not always work and not resilient to changing lock file)`.

Instead, only cache the `~/.pnpm-store` (global cache) and `pnpm install --frozen-lockfile` each time to reconstruct node_modules ( takes a bit time but efficient and we have very few dependencies at the moment to rebuild) which allows for accomadating minor changes for lock files during PR updates and reduces the cache invalidation frequently

For most use cases with PNPM:
- Refer: https://github.com/orgs/pnpm/discussions/6374 
- Cache only ~/.pnpm-store: This is the default and preferred approach for CI pipelines.

Bottomline: Remove node_modules caching for now.